### DISCUSSION TITLE
What's the performance like or the live transcription on a real device?

### DISCUSSION THREAD

**Discussion by chen-rn** [mybigday/whisper.rn#19] [General]
Hey!

Super excited to see this project! I'm going to try and run the example demo when I have some time in the upcoming days.

In the meanwhile, I was wondering if you know off the top of your mind what the performance for the real-time transcription is like, on a modern device!

Any video demos?

**Comment by jhen0409** [comment]
More documentation or demos will be available in the README, they may be later. I've recorded a one minute demo so you can watch it first:

https://user-images.githubusercontent.com/3001525/229002826-80683c3b-50da-46c8-8e87-d68041481cb7.MP4

> iPhone 13 Pro Max (4 threads, `base.en` model) transcribed from a snippet of [Introducing GPT-4](https://www.youtube.com/watch?v=--khbXchTeE) video)

Also, I've moved this issue to Discussions.

**Comment by chefexperte** [comment]
For some reason on my Pocophone F1 it takes over a minute to process a second of recording, which is absolutely terrible and not realtime by any means.

**Reply by jhen0409** [reply]
Since there isn't much info here, I'm not really sure what's going on, but based on the Pocophone F1 specs I searched, I don't think it's that slow. It may be helpful to check [what type of lib is loaded](https://github.com/mybigday/whisper.rn/blob/main/android/src/main/java/com/rnwhisper/WhisperContext.java#L436-L444).

In my limited testing on Pixel 6, using the tiny.en model in release mode, each transcription just took ~2s.

Also, currently this package is only supported run on CPU for Android, unlike iOS has Accelerate Framework / CoreML support. It maybe support GPU (CLBlast for some devices? or Vulkan?) or NNAPI in the future.

**Reply by Manancode** [reply]
Anyone tried doing the same on any android device and achieved lower latency than <2 secs?

**Reply by chefexperte** [reply]
@Manancode I haven't used this project anymore, but there is an open source voice input (they also have a keyboard together with it) that uses Whisper for speech to text and it works quite well. I also have a Pixel 7 now so I can't compare the performance, but it's pretty fast. I use it on a daily basis now. The App is called FUTO Keyboard, it's part of an open source project thingy by Louis Rossman if you know that guy.

**Reply by Manancode** [reply]
@chefexperte does shepra have multi language support like whisper or only few popular languages support?

**Reply by chefexperte** [reply]
I don't know what shepra is, but the keyboard with voice input I mentioned does use normal Whisper and this supports all languages it also supports
